### PR TITLE
Remove custom `MAVEN_OPTS` in favor of `.mvn/jvm.config`

### PR DESCRIPTION
**Pull request by basil**
We set a custom value for `-Xmx` for CI builds in `Jenkinsfile`, overriding the value in `.mvn/jvm.config` that would otherwise be used for both local and CI builds. I see no reason to deviate from the value used in local builds when running in CI, so this PR removes the customization from the `Jenkinsfile`. As a result, the CI builds should use the same value used in local builds, which we should see in `-XX:+PrintCommandLineFlags` output from this CI run. I have never had a problem with the tighter value from local builds, and if anything I think it would benefit CI builds (by reducing memory usage) rather than harming them.

### Testing done

From [the CI build](https://ci.jenkins.io/job/Core/job/jenkins/job/PR-6494/1/consoleFull):

```
-XX:InitialHeapSize=125000000 -XX:MaxHeapSize=838860800 -XX:+PrintCommandLineFlags -XX:+UseCompressedClassPointers -XX:+UseCompressedOops -XX:+UseParallelGC
```

The value of `-XX:MaxHeapSize` (838860800 bytes = 800 MiB) matches the setting in `.mvn/jvm.config` (`-Xmx800m`).

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
